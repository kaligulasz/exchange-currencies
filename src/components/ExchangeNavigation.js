import React from 'react';
import { withRouter } from 'react-router-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import ReactRouterPropTypes from 'react-router-prop-types';

// Components
import Button from './Button';
import CurrencySelectors from './CurrencySelectors';

const Background = styled.header`
  background: ${props => props.theme.color.primary};
  height: 12%;
  display: flex;
`;

const AppWrapper = styled.div`
  width: ${props => props.theme.mobileWidth};
  margin: 0 auto;
  display: flex;
  align-items: center;
  max-width: ${props => props.theme.primaryMaxWidth};
  justify-content: space-between;
  
  @media (min-width: ${props => props.theme.mediaQuery.phoneBig}) {
    width: ${props => props.theme.primaryWidth};
  }
`;

const ExchangeNavigation = ({ history, currencyList }) => {
  const handleOnClick = () => {
    history.push('/');
  };

  return (
    <Background>
      <AppWrapper>
        <CurrencySelectors currencyList={currencyList} />
        <Button onClick={handleOnClick} tertiary uppercase>back</Button>
      </AppWrapper>
    </Background>
  );
};

ExchangeNavigation.propTypes = {
  history: ReactRouterPropTypes.history.isRequired,
  currencyList: PropTypes.array.isRequired,
};

export default withRouter(ExchangeNavigation);
