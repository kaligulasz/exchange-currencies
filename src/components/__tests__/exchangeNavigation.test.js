import React from 'react';
import { shallow } from 'enzyme';

import ExchangeNavigation from '../ExchangeNavigation';

describe('<ExchangeNavigation />', () => {
  it('renders ExchangeNavigation', () => {
    const component = shallow(<ExchangeNavigation />);

    expect(component).toMatchSnapshot();
  });
});
