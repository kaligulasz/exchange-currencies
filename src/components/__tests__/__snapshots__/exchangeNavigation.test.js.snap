// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExchangeNavigation /> renders ExchangeNavigation 1`] = `
<Route>
  <Component />
</Route>
`;
